pkg/utils/gitops/environments.go

```diff
@@ -33,25 +34,33 @@ func (g *GitopsController) GetEnvironmentsList(namespace string) (*appservice.En
 
 // GetEphemeralEnvironment returns the Ephemeral Environment in the namespace and nil if it's not found
 // It will search for the Environment based on the Snapshot and Scneario name present in its labels,
-// and also look for environment containing the tag "ephemeral".
+// and also look for environment containing the "ephemeral" tag.
+// In case of failure, this function retries till it gets timed out.
```

Comment:
Added a retry logic within this function because we saw some infra-error while trying to list the envs.